### PR TITLE
Add doc for deployment of namespaced federation

### PR DESCRIPTION
**Pull request by marun**
Targets #197

### CODE REVIEW

File: docs/userguide.md [hunk 1]
```diff
@@ -323,3 +323,40 @@ The argument(s) used is/are the list of context names of the additional clusters
 unjoined from this federation control plane. Clarifying, say the `host-cluster-context` used is
 `cluster1`, then on successful completion of the script used in example, both `cluster1` and `cluster2`
 will be unjoined from the deployed federation control plane.
+
+## Namespaced Federation
```
Comment (gyliu513) [inline docs/userguide.md]: Do we need to update the tile for above to highlight it is `Cluster Level Federation`?

Including title for `Automated Deployment` and `Manual Deployment`?

Comment (marun) [reply docs/userguide.md]: I don't think I want to rewrite the other sections just yet.  Likely that will happen eventually, but initially the goal is making this feature available more than changing general perception of cluster-scoped federation.

Comment (gyliu513) [reply docs/userguide.md]: I think we do not need to rewrite the other sections, but just updating the title as follows should be good enough currently:

Automated Deployment -> Cluster Level Federation Automated Deployment
Manual Deployment -> Cluster Level Federation Manual Deployment

**Comment by marun** [comment]
I've addressed @font's feedback and added a new section on how to join a cluster to a namespaced federation.

File: docs/userguide.md [hunk 2]
```diff
@@ -323,3 +323,60 @@ The argument(s) used is/are the list of context names of the additional clusters
...
+plane to.  The control plane will only have permission to access this
+on both the host and member clusters.
+
+It may be useful to supply `FEDERATION_NAMESPACE=test-namespace` to
+allow the examples to work unmodified.
+
+### Joining Clusters
+
+Joining additional clusters to a namespaced federation requires
+providing additional arguments to `kubefed join`:
```
Comment (pmorie) [inline docs/userguide.md]: nit: `kubefed2`

**Pull request merged**